core/src/main/java/org/elasticsearch/index/translog/BufferingTranslogWriter.java

```diff
@@ -46,6 +46,7 @@ public BufferingTranslogWriter(ShardId shardId, long generation, ChannelReferenc
 
     @Override
     public Translog.Location add(BytesReference data) throws IOException {
+        ensureOpen();
```

Comment:
well - we have the parent one under lock. I am tempted to remove the one here altogether or just keep them